$NetBSD: patch-Makefile,v 1.1 2018/12/17 14:22:52 bsiegert Exp $

 - Make curses flags configurable
 - Also use pkg-config for regular ncurses
 - Make optimization flags configurable

Upstream PR:
https://github.com/jarun/nnn/pull/166

--- Makefile
+++ Makefile
@@ -6,16 +6,24 @@ STRIP ?= strip
 PKG_CONFIG ?= pkg-config
 INSTALL ?= install
 
-CFLAGS ?= -O3
-CFLAGS += -Wall -Wextra -Wno-unused-parameter
+CFLAGS_OPTIMIZATION ?= -O3
 
 ifeq ($(shell $(PKG_CONFIG) ncursesw && echo 1),1)
-	CFLAGS += $(shell $(PKG_CONFIG) --cflags ncursesw)
-	LDLIBS += $(shell $(PKG_CONFIG) --libs   ncursesw)
+	CFLAGS_CURSES ?= $(shell $(PKG_CONFIG) --cflags ncursesw)
+	LDLIBS_CURSES ?= $(shell $(PKG_CONFIG) --libs   ncursesw)
+else ifeq ($(shell $(PKG_CONFIG) ncurses && echo 1),1)
+	CFLAGS_CURSES ?= $(shell $(PKG_CONFIG) --cflags ncurses)
+	LDLIBS_CURSES ?= $(shell $(PKG_CONFIG) --libs   ncurses)
 else
-	LDLIBS += -lncurses
+	LDLIBS_CURSES ?= -lncurses
 endif
 
+CFLAGS += -Wall -Wextra -Wno-unused-parameter
+CFLAGS += $(CFLAGS_OPTIMIZATION)
+CFLAGS += $(CFLAGS_CURSES)
+
+LDLIBS += $(LDLIBS_CURSES)
+
 DISTFILES = src nnn.1 Makefile README.md LICENSE
 SRC = src/nnn.c
 BIN = nnn
